src/google/protobuf/extension_set.cc

```diff
@@ -1950,7 +1970,7 @@ void ExtensionSet::GrowCapacity(size_t minimum_new_capacity) {
 
   auto new_flat_capacity = flat_capacity_;
   do {
-    new_flat_capacity = new_flat_capacity == 0 ? 1 : new_flat_capacity * 4;
```

Comment:
I think we should stick with the 4x growth, because as I understand this was a deliberate choice to optimize performance compared to the usual 2x growth.

Reply:
understood.